presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp

```diff
@@ -567,9 +571,20 @@ void PeriodicTaskManager::updateSpillStatsTask() {
       kCounterSpillFlushTimeUs, deltaSpillStats.spillFlushTimeUs);
   REPORT_IF_NOT_ZERO(
       kCounterSpillWriteTimeUs, deltaSpillStats.spillWriteTimeUs);
-  if (deltaSpillStats.spilledBytes > 0) {
-    LOG(INFO) << "Spill Stats:" << deltaSpillStats.toString();
+  if (!deltaSpillStats.empty()) {
+    LOG(INFO) << "Updated spill stats: " << updatedSpillStats.toString();
+    LOG(INFO) << "Spill stats change:" << deltaSpillStats.toString();
   }
+
+  const auto spillMemoryStats = velox::exec::Spiller::pool()->stats();
+  LOG(INFO) << "Spill memory usage: current["
+            << velox::succinctBytes(spillMemoryStats.currentBytes) << " peak[{"
+            << velox::succinctBytes(spillMemoryStats.peakBytes) << "}]";
+  REPORT_ADD_STAT_VALUE(
```

Comment:
REPORT_IF_NOT_ZERO ?

Reply:
We shall return no matter empty or not. Otherwise, the monitor will keep seeing the old value?

Reply:
Monitor will not see the old value. Each datapoint should be associated with a timestamp. And when you don't report ODS will automatically fill 0.